template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/impala/ImpalaVolumeConfigProvider.java

```diff
@@ -38,20 +40,51 @@ public List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, HostgroupV
                         .mapToInt(volume -> volume.getVolumeSize())
                         .min()
                         .orElse(0);
+                int volumeCount = 0;
+                int temporaryStorageVolumeCount = 0;
+                int temporaryStorageVolumeSize = 0;
+                if (hostGroupView != null) {
+                    volumeCount = hostGroupView.getVolumeCount();
+                    if (hostGroupView.getTemporaryStorageVolumeCount() != null) {
+                        temporaryStorageVolumeCount = hostGroupView.getTemporaryStorageVolumeCount();
+                    }
+                    if (hostGroupView.getTemporaryStorageVolumeSize() != null) {
+                        temporaryStorageVolumeSize = hostGroupView.getTemporaryStorageVolumeSize();
+                    }
+                }
 
                 List<ApiClusterTemplateConfig> configs = new ArrayList<>();
-                configs.add(config(IMPALA_SCRATCH_DIRS_PARAM,
-                        buildVolumePathStringZeroVolumeHandled(hostGroupView.getVolumeCount(), "impala/scratch")));
+
+                if (checkTemporaryStorage(hostGroupView, temporaryStorageVolumeCount)) {
+                    configs.add(config(IMPALA_SCRATCH_DIRS_PARAM,
+                            buildEphemeralVolumePathString(temporaryStorageVolumeCount, "impala/scratch")));
+                } else {
+                    configs.add(config(IMPALA_SCRATCH_DIRS_PARAM,
+                            buildVolumePathStringZeroVolumeHandled(volumeCount, "impala/scratch")));
+                }
 
                 if (minAttachedVolumeSize > MIN_ATTACHED_VOLUME_SIZE_TO_ENABLE_CACHE_IN_GB
                         && hostGroupView.getVolumeCount() > 0) {
```

Comment:
`volumeCount` is already extracted above

Reply:
Yes, I will change it here also.